src/hazmat/float.rs

```diff
@@ -113,12 +118,176 @@ pub(crate) const fn floor_sqrt(x: u32) -> u32 {
     (low + high) / 2
 }
 
+// Copied from the `libm` crate, which is licensed under the MIT license.
+
+/* origin: FreeBSD /usr/src/lib/msun/src/e_log2.c */
+/*
+ * ====================================================
+ * Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+ *
+ * Developed at SunSoft, a Sun Microsystems, Inc. business.
+ * Permission to use, copy, modify, and distribute this
+ * software is freely granted, provided that this notice
+ * is preserved.
+ * ====================================================
+ */
+/*
+ * Return the base 2 logarithm of x.  See log.c for most comments.
+ *
+ * Reduce x to 2^k (1+f) and calculate r = log(1+f) - f + f*f/2
+ * as in log.c, then combine and scale in extra precision:
+ *    log2(x) = (f - f*f/2 + r)/log(2) + k
+ */
+
+/// The base 2 logarithm of `x` (f64).
+#[allow(clippy::all)]
```

Comment:
How many exceptions are actually needed here? Perhaps it's better to just enumerate them explicitly?

Reply:
A LOT. There are multiple screens full of warnings. I did not want to make any changes to the code to make it easy to maintain, but it's a port of C code that adhere to practices common in C (likely to make the Rust port easy to maintain&compare to the original?).

Given that it's code we should not ever need to touch, beyond re-vendoring it if `libm` ever changes it, I think it's fine.